src/trio/_core/_parking_lot.py

```diff
@@ -234,6 +269,24 @@ def repark_all(self, new_lot: ParkingLot) -> None:
         """
         return self.repark(new_lot, count=len(self))
 
+    def break_lot(self, task: Task | None = None) -> None:
+        """Break this lot, causing all parked tasks to raise an error, and any
+        future tasks attempting to park (and unpark? repark?) to error. The error
+        contains a reference to the task sent as a parameter.
+        """
+        if task is None:
+            task = _core.current_task()
+        self.broken_by = task
```

Comment:
I'm just now realizing that this relationship is a bit flawed. Lots can only be marked as broken by one task, but multiple tasks could be marked as a parking lot breaker and exit (or a task multiple times, but that's handled already). I'm not sure what's a good idea. We can't raise any error because there's no good place to raise errors, but I suppose we could warn? Or maybe `broken_by` should be a list.

Reply:
should warning be raised if the same task breaks a lot multiple times?

Reply:
Yeah that sounds like a fine warning and a task breaking a lot multiple times is probably fine. After all we allow nesting these add/remove_parking_lot_breaker